src/oscar/static/oscar/js/oscar/dashboard.js

```diff
@@ -275,6 +275,12 @@ var oscar = (function(o, $) {
                 if (!form || $(":invalid", form).length == 0)
                     $(this).button('loading');
             });
+
+            // Display tabs that have invalid input fields
+            $('input').on('invalid', function(){
+                var id = $(this).closest('.tab-pane').attr('id');
+                $('.nav-list a[href="#' + id + '"]').tab('show');
```

Comment:
Could we add a check here that `id` isn't `undefined` before attempting to show the tab? This code executes for all dashboard forms, including those (e.g., voucher creation form) that don't have a tabbed interface.

Reply:
We could but it does not trigger an error at the moment. jQuery returns an empty jQuery collection when `id` is `undefined`.

Reply:
The string concatenation means that the second line looks for a DOM element with `href="#undefined`". While this doesn't raise an error I would prefer not to rely on there being no such element in the DOM - it's a potential seed for confusion when unrelated things are changed in future.

Reply:
Fair point :)